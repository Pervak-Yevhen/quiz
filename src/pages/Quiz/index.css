.Quiz {
    position: relative;
    flex: 1;
    display: flex;
    justify-content: center;
}

.Quiz__content {
    flex: 3;
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    padding: 133px 20px 122px 80px;
    background-color: var(--cl-black-5);
    font: var(--font-base);
}

.Quiz__content-title {
    max-width: 624px;
    margin: 0 0 20px 0;
    font: var(--font-question);
}

.Quiz__panel {
    flex: 1;
    display: flex;
    align-items: center;
}

@media (max-width: 760px) {
    .Quiz__content {
        flex: 1;
        padding: 16px 0;
        justify-content: space-around;
        align-items: center;
    }

    .Quiz__content-title {
        text-align: center;
        padding: 0 10px;
    }

    .Quiz__panel {
        z-index: 1;
        position: absolute;
        width: 0;
        right: 0;
        bottom: 0;
        top: 0;
        overflow: hidden;
        transition: width .3s;
    }

    .Quiz__panel--show {
        width: 100%;
        background-color: var(--cl-black-5);
    }
    .Quiz__button {
        position: absolute;
        width: 24px;
        height: 24px;
        z-index: 2;
        top: 20px;
        right: 20px;
        color: inherit;
        border: none;
        padding: 0;
        cursor: pointer;
        outline: inherit;
        background-image: url('/src/assets/svgs/menu.svg');
    }
    .Quiz__button--active {
        background-image: url('/src/assets/svgs/close.svg');
    }
}
