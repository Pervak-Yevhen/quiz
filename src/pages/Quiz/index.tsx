import React, { useState, useEffect } from 'react';
import cn from "classnames";
import Item from '../../componets/List/Item';
import List from '../../componets/List';
import ScorePanel from "../../componets/ScorePanel";
import { useQuizContext } from "../../context/quiz-context";
import { useQuizScoreContext } from "../../context/quiz-score-context";
import { useResizeContext } from "../../context/resize-context";

import './index.css'

const Quiz = () => {
  const [showPanel, setState] = useState(false);
  const {
    item: { question, variants, answer },
    onItemClick,
    wrongAnswerSelected,
    selectedItemIndex,
  } = useQuizContext();

  const { list, questionIndex } = useQuizScoreContext();
  const isMobile = useResizeContext();
  const onShowPanel = () => setState(!showPanel);
  useEffect(() => {
    if (isMobile) setState(false);
    else setState(true);
  }, [isMobile]);

  const button = (
    <button
      className={cn('Quiz__button', {
        "Quiz__button--active": showPanel
      })}
      onClick={onShowPanel}
    />
  );

  const checkIsRight = (index: number):boolean => {
    const selected = index === selectedItemIndex;
    let isRight = false;
    if (selectedItemIndex != null) isRight = answer.includes(selectedItemIndex) && selected;
    return isRight || (wrongAnswerSelected && answer.includes(index));
  };

  return (
    <div className="Quiz">
      { isMobile && button }
      <div className="Quiz__content">
        <h2 className="Quiz__content-title">{question}</h2>
        <List>
          {
            variants.map((text: string, i: number) => {
              const selected = i === selectedItemIndex;
              return (
                <Item
                  key={i}
                  selected={selected}
                  right={checkIsRight(i)}
                  wrong={selected && wrongAnswerSelected}
                  text={text}
                  onItemClick={() => onItemClick(i)}
                  withAlphabeticCount
                />
              );
            })
          }
        </List>
      </div>
      <div className={cn('Quiz__panel', {
        'Quiz__panel--show': showPanel && isMobile,
      })}
      >
        <ScorePanel list={list} currentItem={questionIndex} />
      </div>
    </div>
  );
}

export default Quiz;
