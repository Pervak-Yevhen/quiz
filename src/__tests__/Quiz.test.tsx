import React from 'react';
import { render } from '@testing-library/react';
import {BrowserRouter} from 'react-router-dom'
import Quiz from "../pages/Quiz";

test('Quiz render', () => {
  const { asFragment } = render(<Quiz />, { wrapper: BrowserRouter });
  expect(asFragment()).toMatchSnapshot();
});
