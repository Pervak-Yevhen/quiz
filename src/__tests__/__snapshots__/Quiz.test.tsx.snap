// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quiz render 1`] = `
<DocumentFragment>
  <div
    class="Quiz"
  >
    <div
      class="Quiz__content"
    >
      <h2
        class="Quiz__content-title"
      />
      <ul
        class="List"
      />
    </div>
    <div
      class="Quiz__panel"
    >
      <div
        class="ScorePanel"
      >
        <ul
          class="List List--column"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
